### PR TITLE
Update interops for dev17

### PR DESCRIPTION
**Pull request by dmonroym**
Related to [AB#1297116](https://devdiv.visualstudio.com/0bdbc590-a062-4c3f-b0f6-9383f67865ee/_workitems/edit/1297116)

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/project-system/pull/7057)

### CODE REVIEW

**Review by jjmew — APPROVED** [review]

File: build.cmd
Comment (drewnoakes) [inline build.cmd:60]: Let's make sure we bring this back in the future. Can we file a bug about this?

File: build/import/HostAgnostic.props
Comment (drewnoakes) [inline build/import/HostAgnostic.props:33]: Do we no longer need the `.SDK` package?

File: src/Microsoft.VisualStudio.Editors/Common/Utils.vb
Comment (drewnoakes) [inline src/Microsoft.VisualStudio.Editors/Common/Utils.vb:1350]: Can we log an issue against 17.0 to make sure we don't forget to revert all the occurrences of `TODO: Merged PIAs` in the code?

**Review by drewnoakes — COMMENTED** [review]
Just a question and request to log an issue covering all the things we need to remember to reinstate in future.

**Comment by dmonroym** [comment]
@drewnoakes there's still a decent amount of work to be done here as we depend on packages that are still getting updated and this builds with over 150 warnings. This PR only serves to unblock our partners but wouldn't meet the requirements to get merged into the VS repo yet so treat the PRs going into the branch as Drafts

File: build/import/HostAgnostic.props
Comment (dmonroym) [reply build/import/HostAgnostic.props:33]: Having these packages is the equivalent to it but hope is to bring it back. Simply waiting for that package to be updated. For now using what's available

File: src/Microsoft.VisualStudio.Editors/Common/Utils.vb
Comment (dmonroym) [reply src/Microsoft.VisualStudio.Editors/Common/Utils.vb:1350]: Won't be able to forget since the PR won't merge without these :)

File: src/Microsoft.VisualStudio.AppDesigner/Common/ShellUtil.vb
Comment (MiYanni) [inline src/Microsoft.VisualStudio.AppDesigner/Common/ShellUtil.vb:189]: Just for my understanding, why did a bunch of these need to use explicit `get_` calls?

Comment (dmonroym) [reply src/Microsoft.VisualStudio.AppDesigner/Common/ShellUtil.vb:189]: I kept seeing these errors:
```
G:\project-system\src\Microsoft.VisualStudio.AppDesigner\Common\ShellUtil.vb(189,45): error BC30456: 'Properties' is not a member of 'DTE'. [G:\project-system\src\Microsoft.VisualStudio.AppDesigner\Microsoft.VisualStudio.AppDesigner.vbproj]
G:\project-system\src\Microsoft.VisualStudio.AppDesigner\PropPages\PropPageUserControlBase.vb(2836,20): error BC30456: 'FileNames' is not a member of 'ProjectItem'. [G:\project-system\src\Microsoft.VisualStudio.AppDesigner\Microsoft.VisualStudio.AppDesigner.vbproj]
```
Ryan Molden then replied with this: Properties and FileNames are both named indexers, they were converted to get_Properties and get_FileNames in the PIA you are consuming. There is ongoing work to provide a way to NOT have to convert named indexers to get_ equivalents, but it is what must be done today.

**Pull request merged**
